typescript_test/test.ts

```diff
@@ -580,6 +580,104 @@ function testArrayArgument() {
   }
 }
 
+function testOptionalArgumentsConflicting() {
+  type State = { foo: string; bar: number; baz: boolean }
+
+  const selector = createSelector(
+    (state: State) => state.baz,
+    (state: State, arg: string) => arg,
+    (state: State, arg: number) => arg,
+    (baz) => {
+      const baz1: boolean = baz
+      // @ts-expect-error
+      const baz2: number = baz
+    }
+  )
+
+  // @ts-expect-error the selector above has inconsistent conflicting arguments so usage should error
```

Comment:
on master there is no error - even though we expect a parameter, we are allowed to call it without one